docs/madr/decisions/055-default-hostname-generators.md

```diff
@@ -0,0 +1,218 @@
+
+This HostnameGenerator could be automatically created by Universal Zone CP.
+
+#### Synced Universal MeshService
+
+```yaml
+spec:
+  template: '{{ .DisplayName }}.svc.mesh.{{ label "kuma.io/zone" }}'
+  selector:
+    meshService:
+      matchLabels:
+        kuma.io/origin: global
+        kuma.io/env: universal # short term the users need to apply it themselves 
+```
+
+This HostnameGenerator could be automatically created by Global CP so that it's synced down to every Zone CP.
+
+### MeshExternalService
+
+#### MeshExternalService applied on Zone CP
+
+```yaml
+spec:
+  template: '{{ .DisplayName }}.svc.meshext.local'
+  selector:
+    meshExternalService:
+      matchLabels:
+        kuma.io/origin: zone # only consider local MeshExternalServices
+```
+
+This HostnameGenerator could be automatically created by Zone CP.
+
+On Kubernetes, MeshExternalService can only be applied in the system namespace, so for example hostname for this MeshExternalService
+
+```yaml
+kind: MeshExternalService
+metadata:
+  name: httpbin
+  namespace: kuma-system
+```
+
+Would be `httpbin.svc.meshext.local`
+
+#### MeshExternalService applied on Global CP
+
+```yaml
+spec:
+  template: '{{ .DisplayName }}.svc.meshext.global'
+  selector:
+    meshExternalService:
+      matchLabels:
+        kuma.io/origin: global # only consider synced MeshExternalServices
+```
+
+This HostnameGenerator could be automatically created by Global CP.
+
+### MeshMultizoneService 
+
+#### MeshMultizoneService applied on Global CP
+
+```yaml
+spec:
+  template: '{{ .DisplayName }}.svc.meshmz.global'
+  selector:
+    kuma.io/origin: global # only consider synced MeshMultizoneService
+```
+
+This HostnameGenerator could be automatically created by Global CP.
+
+#### MeshMultizoneService applied on Zone CP
+
+The plan is to only allow applying MeshMultizoneService on Global CP. However, if we want to extend this so it's also possible to apply it on Zone CP, we can do this
+
+```yaml
+spec:
+  template: '{{ .DisplayName }}.svc.meshmz.local'
+  selector:
+    kuma.io/origin: zone # only consider local MeshMultizoneService
+```
+
+This HostnameGenerator could be automatically created by Zone CP.
+
+## Pros and Cons of the Options - Shipping
+
+### Documenting
+
+We can just document HostnameGenerators.
+
+### Creating
+
+We can create HostnameGenerators for users. The what and where is described in the "Pros and Cons of the Options - Template and matching"
+
+## Pros and Cons of the Options - Suffix
+
+### `.svc.{{ type }}.{{ zone }}`
+
+Types:
+* `mesh` for MeshService
+* `meshext` - MeshExternalService
+* `meshmz` - MeshMultizoneService
+
+Examples:
+* `redis.kuma-demo.mesh.east`
+* `backend.west.mesh.west`
+* `httpbin.meshext.local`
+* `google.meshext.global`
+* `auth.meshmz.global`
+
+Advantages:
+* It's familiar to Kubernetes pattern of `svc.cluster.local`
+
+Disadvantages:
+* `{{ zone }}` might violate TLD. Although we can validate zone name to prevent this.
```

Comment:
do we validate zone name in any way? would this be painful to migrate?

Reply:
```
func (r *ZoneConfig) Validate() error {
	if r.Name == "" {
		return errors.Errorf("Name is mandatory")
	}
	if !govalidator.IsDNSName(r.Name) {
		return errors.Errorf("Zone name %s has to be a valid DNS name", r.Name)
	}
	if len(r.Name) > 63 {
		return errors.New("Zone name cannot be longer than 63 characters")
	}
```
We do, although I'm not sure yet if `govalidator.IsDNSName` == `valid TLD`

Reply:
It's just a regex 

```
DNSName           string = `^([a-zA-Z0-9_]{1}[a-zA-Z0-9_-]{0,62}){1}(\.[a-zA-Z0-9_]{1}[a-zA-Z0-9_-]{0,62})*[\._]?$
```` 

so I wouldn't say so

Reply:
how hard is renaming a zone? is it just chaning `KUMA_MULTIZONE_ZONE_NAME`?

Reply:
In theory yes, but in practice I suspect you can get some turbulence, because for example DPPs has `kuma.io/zone` tag as well as some policies `kuma.io/zone` label

Reply:
oh, that's right, so let's go with the other one - `.{{ zone }}.{{ type }}`